aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/helper/HeaderUtils.java

```diff
@@ -0,0 +1,321 @@
+     * @param valueSeparator The separator to be used for parsing header values
+     * @return A list of SimpleMapEntry objects with all of the possible values for the header.
+     */
+    public static List<HeaderValue> parseHeaderValue(String headerValue, String valueSeparator, String qualifierSeparator) {
+        // Accept: text/html, application/xhtml+xml, application/xml;q=0.9, */*;q=0.8
+        // Accept-Language: fr-CH, fr;q=0.9, en;q=0.8, de;q=0.7, *;q=0.5
+        // Cookie: name=value; name2=value2; name3=value3
+        // X-Custom-Header: YQ==
+
+        List<HeaderValue> values = new ArrayList<>();
+        if (headerValue == null) {
+            return values;
+        }
+
+        for (String v : headerValue.split(valueSeparator)) {
+            String curValue = v;
+            float curPreference = 1.0f;
+            HeaderValue newValue = new HeaderValue();
+            newValue.setRawValue(v);
+
+            for (String q : curValue.split(qualifierSeparator)) {
+
+                String[] kv = q.split(HEADER_KEY_VALUE_SEPARATOR, 2);
+                String key = null;
+                String val = null;
+                // no separator, set the value only
+                if (kv.length == 1) {
+                    val = q.trim();
+                }
+                // we have a separator
+                if (kv.length == 2) {
+                    // if the length of the value is 0 we assume that we are looking at a
+                    // base64 encoded value with padding, so we just set the value. This is because
+                    // we assume that empty values in a key/value pair will contain at least a white space
+                    if (kv[1].length() == 0) {
+                        val = q.trim();
+                    }
+                    // this was a base64 string with an additional = for padding, set the value only
+                    if ("=".equals(kv[1].trim())) {
+                        val = q.trim();
+                    } else { // it's a proper key/value set both
+                        key = kv[0].trim();
+                        val = ("".equals(kv[1].trim()) ? null : kv[1].trim());
+                    }
+                }
+
+                if (newValue.getValue() == null) {
+                    newValue.setKey(key);
+                    newValue.setValue(val);
+                } else {
+                    // special case for quality q=
+                    if ("q".equals(key)) {
+                        curPreference = Float.parseFloat(val);
+                    } else {
+                        newValue.addAttribute(key, val);
+                    }
+                }
+            }
+            newValue.setPriority(curPreference);
+            values.add(newValue);
+        }
+
+        // sort list by preference
+        values.sort((HeaderValue first, HeaderValue second) -> {
+            if ((first.getPriority() - second.getPriority()) < .001f) {
```

Comment:
That's an epsilon. Instead of comparing 2 float values for equality (with ==), we check whether their difference is within some error bounds or epsilon value, like this: `bool isEqual = fabs(f1 – f2) <= epsilon;` This calculation allows us to express the concept of two floats being close enough that we want to consider them to be equal. [Ref](https://randomascii.wordpress.com/2012/02/25/comparing-floating-point-numbers-2012-edition/)